src/object.ts

```diff
@@ -29,12 +29,12 @@ import type { DeepMerge } from './types'
  * // { b: 2 }
  * ```
  */
-export function objectMap<K extends string, V, NK = K, NV = V>(obj: Record<K, V>, fn: (key: K, value: V) => [NK, NV] | undefined): Record<K, V> {
+export function objectMap<K extends string, V, NK extends string | number | symbol = K, NV = V>(obj: Record<K, V>, fn: (key: K, value: V) => [NK, NV] | undefined): Record<NK, NV> {
```

Comment:
Just for curiosity, what do you think using of `PropertyKey` instead of `string | number | symbol`.

Reply:
> Just for curiosity, what do you think using of `PropertyKey` instead of `string | number | symbol`?

This idea of yours is better I think. I use `string | number | symbol` just because that's what VSCode suggests.

Reply:
Thanks for the honest answer.